cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -29,92 +29,141 @@ import (
+	// BindPort sets the secure port for the API Server to bind to.
+	// Defaults to 6443.
 	BindPort int32
 }
 
-// TokenDiscovery contains elements needed for token discovery
+// TokenDiscovery contains elements needed for token discovery.
 type TokenDiscovery struct {
-	ID        string
-	Secret    string
+	// ID is the first part of a bootstrap token. Considered public information.
+	// It is used when referring to a token without leaking the secret part.
+	ID string
+	// Secret is the second part of a bootstrap token. Should only be shared
+	// with trusted parties.
+	Secret string
+	// TODO: Seems unused. Remove?
 	Addresses []string
 }
 
-// Networking contains elements describing cluster's networking configuration
+// Networking contains elements describing cluster's networking configuration.
 type Networking struct {
+	// ServiceSubnet is the subnet used by k8s services. Defaults to "10.96.0.0/12".
 	ServiceSubnet string
-	PodSubnet     string
-	DNSDomain     string
+	// PodSubnet is the subnet used by pods.
+	PodSubnet string
+	// DNSDomain is the dns domain used by k8s services. Defaults to "cluster.local".
+	DNSDomain string
 }
 
-// Etcd contains elements describing Etcd configuration
+// Etcd contains elements describing Etcd configuration.
 type Etcd struct {
+	// Endpoints of etcd members. Useful for using external etcd.
+	// If not provided, kubeadm will run etcd in a static pod.
 	Endpoints []string
-	CAFile    string
-	CertFile  string
-	KeyFile   string
-	DataDir   string
+	// CAFile is an SSL Certificate Authority file used to secure etcd communication.
+	CAFile string
+	// CertFile is an SSL certification file used to secure etcd communication.
+	CertFile string
+	// KeyFile is an SSL key file used to secure etcd communication.
+	KeyFile string
+	// DataDir is the directory etcd will place its data.
+	// Defaults to "/var/lib/etcd".
+	DataDir string
+	// ExtraArgs are extra arguments provided to the etcd binary
+	// when run inside a static pod.
 	ExtraArgs map[string]string
-	// Image specifies which container image to use for running etcd. If empty, automatically populated by kubeadm using the image repository and default etcd version
-	Image      string
+	// Image specifies which container image to use for running etcd.
+	// If empty, automatically populated by kubeadm using the image
+	// repository and default etcd version.
+	Image string
+	// SelfHosted holds configuration for self-hosting etcd.
 	SelfHosted *SelfHostedEtcd
 }
 
-// SelfHostedEtcd describes options required to configure self-hosted etcd
+// SelfHostedEtcd describes options required to configure self-hosted etcd.
 type SelfHostedEtcd struct {
-	// CertificatesDir represents the directory where all etcd TLS assets are stored. By default this is
-	// a dir names "etcd" in the main CertificatesDir value.
+	// CertificatesDir represents the directory where all etcd TLS assets are stored.
+	// Defaults to "/etc/kubernetes/pki/etcd".
```

Comment:
Is it necessary to indicate the default value of such fields? If the consts of these default values, e.g. `DefaultEtcdCertDir`, change (though they may not) and we forget to update the comment here, it will lead to inconsistency. @kargakis WDYT?

Reply:
Defaults do not change in the same version.

Reply:
New version means new types so these comments need to be vetted anyway.

Reply:
Is it possible that we don't change the type but change the below const between versions? If so, `Defaults to "/etc/kubernetes/pki/etcd"` is a "hard-coded" comment.

https://github.com/kubernetes/kubernetes/blob/29aff5bf47132ee40d3ff1e9bf508acf21a712f7/cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go#L60-L61

Reply:
That constant should not change otherwise we are breaking existing users. Newer versions should use their own constants.

Btw, we "hardcode" defaults in comments elsewhere too.
https://github.com/kubernetes/api/blob/65f67c9cb59d466d2eb758ad9d5ad016eb43c2aa/apps/v1/types.go#L109
Since defaults are API, there is no reason to hide them in constants.

Reply:
agreed :)